shared/common-adapters/avatar.desktop.js

```diff
@@ -17,20 +18,42 @@ type State = {
 const _avatarCache: {[key: string]: ?boolean} = {
 }
 
-class Avatar extends PureComponent<void, Props, State> {
+class Avatar extends Component<void, Props, State> {
   state: State;
 
   constructor (props: Props) {
     super(props)
-    this.state = {avatarLoaded: false, errored: false, url: shared.createAvatarUrl(props)}
+
+    const url = shared.createAvatarUrl(props)
+
+    this.state = {
+      ...this._getLoadedErrorState(url),
+    }
+  }
+
+  _getLoadedErrorState (url: ?string) {
+    return {
+      avatarLoaded: !!url && _avatarCache.hasOwnProperty(url) && !!_avatarCache[url],
+      errored: !!url && _avatarCache.hasOwnProperty(url) && !_avatarCache[url],
+      url,
+    }
+  }
+
+  shouldComponentUpdate (nextProps: Props, nextState: State) {
+    return !shallowEqual(this.state, nextState) || !shallowEqual(this.props, nextProps)
```

Comment:
yeah its possible we would re-render if the styles change but i wasn't seeing that happen much in my profiling. it won't be buggy it just might re-render when it doesn't need to